File: Python/webhookServer.py
# Author: Hussain Al Zerjawi
# Purpose: Practicing webhooks using Python and the Flask framework
# Date: 12/10/2022

from flask import Flask, request, abort, Response

# Initialize the app variable with a Flask instance
app = Flask(__name__)


@app.route("/webhook", methods=["POST"])
def webhook():
    if request.method == "POST":
        print(request.json)
        return Response(status=200)
    else:
        abort(400)

if __name__ == '__main__':
    app.run()

File: NodeJS/webhookServer.js
// Author: Hussain Al Zerjawi
// Purpose: Practicing webhooks using JavaScript/NodeJS
// Date: 12/10/2022

// Require express and body-parser
const express = require("express")
const bodyParser = require("body-parser")

// Initialize express and define a port
const app = express()
const PORT = 5000

// Tell express to use body-parser's JSON parsing
app.use(bodyParser.json())

// Start express on the defined Port
app.listen(PORT, () => console.log(`Server running on port ${PORT}`))

// app.use(bodyParser.json())
app.post("/webhook", (req, res) => {
    console.log(req.body) // Call your action on the request here
    res.status(200).end() // Responding with a 200 aka status OK
})
